nats.go

```diff
@@ -1904,18 +1909,12 @@ func (nc *Conn) doReconnect(err error) {
 			nc.ach.push(func() { nc.Opts.ReconnectedCB(nc) })
 		}
 
-		lastErr := nc.current.lastErr
-
 		// Release lock here, we will return below.
 		nc.mu.Unlock()
 
 		// Make sure to flush everything
 		nc.Flush()
 
-		if lastErr != nil && !nc.IsClosed() {
```

Comment:
Why was this removed? Not needed? It was originally.

Reply:
I moved the clearing up to line 1874. We need to clear under the lock after processConnectInit() did return OK and before the readLoop has a chance to start. Otherwise you would take the risk of clearing an error that could have been already sent by the server.
This was actually happening with the trimming fix but original test in a Linux VM. I could see the server sending async -ERR but going into reconnect loop because the error was cleared at the wrong place.